server/src/internalClusterTest/java/org/opensearch/search/query/SimpleQueryStringIT.java

```diff
@@ -651,6 +651,53 @@ private void doAssertLimitExceededException(String field, int exceedingFieldCoun
         );
     }
 
+    public void testLimitOnBooleanClauses() throws Exception {
+        String index = "test";
+
+        // LuceneTestCase-TestRuleSetupAndRestoreInstanceEnv resets the max clauses to default, hence initialize with an updated default
+        assertAcked(
+            client().admin()
+                .cluster()
+                .prepareUpdateSettings()
+                .setTransientSettings(Settings.builder().put(INDICES_MAX_CLAUSE_COUNT_SETTING.getKey(), CLUSTER_MAX_CLAUSE_COUNT - 1))
+        );
+        client().prepareIndex(index, "type1", "1").setSource("field1", "foo bar baz").get();
+        refresh();
+
+        // First we create a query with clauses greater than allowed limit
+        StringBuilder sb = new StringBuilder("foo");
+        for (int i = 0; i <= CLUSTER_MAX_CLAUSE_COUNT; i++) {
+            sb.append(" OR foo" + i);
+        }
+        QueryStringQueryBuilder qb = queryStringQuery(sb.toString()).field("field1");
+        SearchPhaseExecutionException e = expectThrows(
+            SearchPhaseExecutionException.class,
+            () -> { client().prepareSearch(index).setQuery(qb).get(); }
+        );
+        logger.info("Exception hit on breaching max clauses for boolean query", e);
+        assert (e.getDetailedMessage().contains("maxClauseCount is set to " + (CLUSTER_MAX_CLAUSE_COUNT - 1)));
+
+        // Update max clause limit and now the same query should execute successfully
+        assertAcked(
+            client().admin()
+                .cluster()
+                .prepareUpdateSettings()
+                .setTransientSettings(Settings.builder().put(INDICES_MAX_CLAUSE_COUNT_SETTING.getKey(), 2048))
+        );
+        Thread.sleep(10);
```

Comment:
Can we check if ```max_clause_count``` value is updated to new value (with smaller sleeps and timeout) before proceeding?

Reply:
done - reduced sleep time from 10 to 1ms (the ideal way is to track via countdown latches but all my local runs pass with even a 1ms sleep so making the simpler change)